### PR TITLE
decrease sample size, install deps before running the tests

### LINKED ISSUE #1755: Hang in DNS concurrency tests
https://ci.vmware.run/vmware/vic/3452

There is a VERY long set of stack traces in the build output. This is a tiny fraction of them as a sample:
```
+ go test -v -covermode=count -coverprofile=/drone/src/github.com/vmware/vic/.cover/github.com-vmware-vic-lib-dns.cover github.com/vmware/vic/lib/dns
=== RUN   TestCache
--- PASS: TestCache (0.00s)
=== RUN   TestCapacity
--- PASS: TestCapacity (0.00s)
=== RUN   TestReset
--- PASS: TestReset (0.00s)
=== RUN   TestExpiration
--- PASS: TestExpiration (0.00s)
=== RUN   TestConcurrency
SIGQUIT: quit
PC=0x461fb1 m=0

goroutine 0 [idle]:
runtime.futex(0x1e11228, 0x0, 0x0, 0x0, 0x0, 0x1e10670, 0x0, 0x0, 0x4109b4, 0x1e11228, ...)
	/usr/local/go/src/runtime/sys_linux_amd64.s:306 +0x21
runtime.futexsleep(0x1e11228, 0x0, 0xffffffffffffffff)
	/usr/local/go/src/runtime/os1_linux.go:40 +0x53
runtime.notesleep(0x1e11228)
	/usr/local/go/src/runtime/lock_futex.go:145 +0xa4
runtime.stopm()
	/usr/local/go/src/runtime/proc.go:1538 +0x10b
runtime.gcstopm()
	/usr/local/go/src/runtime/proc.go:1742 +0x128
runtime.schedule()
	/usr/local/go/src/runtime/proc.go:2039 +0x9c
runtime.goschedImpl(0xc820001800)
	/usr/local/go/src/runtime/proc.go:2155 +0x134
runtime.gopreempt_m(0xc820001800)
	/usr/local/go/src/runtime/proc.go:2170 +0x32
runtime.newstack()
	/usr/local/go/src/runtime/stack.go:932 +0xa8c
runtime.morestack()
	/usr/local/go/src/runtime/asm_amd64.s:359 +0x7f

goroutine 1 [chan receive, 9 minutes]:
testing.RunTests(0x160ed30, 0x1df7d20, 0x7, 0x7, 0xc820074b01)
	/usr/local/go/src/testing/testing.go:583 +0x8d2
testing.(*M).Run(0xc820f3af08, 0x1e102c0)
	/usr/local/go/src/testing/testing.go:515 +0x81
main.main()
	github.com/vmware/vic/lib/dns/_test/_testmain.go:116 +0x253

goroutine 17 [syscall, 9 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1998 +0x1

goroutine 18 [runnable]:
github.com/vmware/vic/lib/dns.(*Cache).Count(0xc8201afa40, 0xc8201b7dd0)
	github.com/vmware/vic/lib/dns/_test/_obj_test/cache.go:70
github.com/vmware/vic/lib/dns.TestConcurrency(0xc8201d1200)
	/drone/src/github.com/vmware/vic/lib/dns/cache_test.go:213 +0x409
testing.tRunner(0xc8201d1200, 0x1df7d80)
	/usr/local/go/src/testing/testing.go:473 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:582 +0x892

goroutine 2223742 [semacquire]:
sync.runtime_Semacquire(0xc82156623c)
	/usr/local/go/src/runtime/sema.go:47 +0x26
sync.(*Mutex).Lock(0xc821566238)
	/usr/local/go/src/sync/mutex.go:83 +0x1c4
sync.(*Pool).Put(0x1df2910, 0x12ad080, 0xc839b292b0)
	/usr/local/go/src/sync/pool.go:81 +0xa5
fmt.(*pp).free(0xc839b292b0)
	/usr/local/go/src/fmt/print.go:149 +0x79
fmt.Sprintf(0x12c8ba8, 0x5, 0xc8e4ce8ea0, 0x2, 0x2, 0x0, 0x0)
	/usr/local/go/src/fmt/print.go:205 +0xc8
github.com/vmware/vic/lib/dns.generateKey(0xc8201d3100, 0x1e, 0x10001, 0x0, 0x0)
	github.com/vmware/vic/lib/dns/_test/_obj_test/cache.go:79 +0x1aa
github.com/vmware/vic/lib/dns.(*Cache).Get(0xc8201afa40, 0xc8201d1290, 0xc8201d45e0)
	github.com/vmware/vic/lib/dns/_test/_obj_test/cache.go:128 +0x7a
github.com/vmware/vic/lib/dns.TestConcurrency.func2()
	/drone/src/github.com/vmware/vic/lib/dns/cache_test.go:190 +0xbe
created by github.com/vmware/vic/lib/dns.TestConcurrency
	/drone/src/github.com/vmware/vic/lib/dns/cache_test.go:210 +0x3e5

goroutine 2031547 [semacquire]:
sync.runtime_Semacquire(0xc8215667bc)
	/usr/local/go/src/runtime/sema.go:47 +0x26
sync.(*Mutex).Lock(0xc8215667b8)
	/usr/local/go/src/sync/mutex.go:83 +0x1c4
sync.(*Pool).Put(0x1df2910, 0x12ad080, 0xc8615a95f0)
	/usr/local/go/src/sync/pool.go:81 +0xa5
fmt.(*pp).free(0xc8615a95f0)
	/usr/local/go/src/fmt/print.go:149 +0x79
fmt.Sprintf(0x12c8ba8, 0x5, 0xc867836ea0, 0x2, 0x2, 0x0, 0x0)
	/usr/local/go/src/fmt/print.go:205 +0xc8
github.com/vmware/vic/lib/dns.generateKey(0xc8201d3100, 0x1e, 0x10001, 0x0, 0x0)
	github.com/vmware/vic/lib/dns/_test/_obj_test/cache.go:79 +0x1aa
github.com/vmware/vic/lib/dns.(*Cache).Get(0xc8201afa40, 0xc8201d1290, 0xc8201d45e0)
	github.com/vmware/vic/lib/dns/_test/_obj_test/cache.go:128 +0x7a
github.com/vmware/vic/lib/dns.TestConcurrency.func2()
	/drone/src/github.com/vmware/vic/lib/dns/cache_test.go:190 +0xbe
created by github.com/vmware/vic/lib/dns.TestConcurrency
	/drone/src/github.com/vmware/vic/lib/dns/cache_test.go:210 +0x3e5

goroutine 3975327 [GC assist wait]:
github.com/vmware/vic/lib/dns.generateKey(0xc8201d3100, 0x1e, 0x10001, 0x0, 0x0)
	github.com/vmware/vic/lib/dns/_test/_obj_test/cache.go:79 +0xf6
github.com/vmware/vic/lib/dns.(*Cache).Get(0xc8201afa40, 0xc8201d1290, 0xc8201d45e0)
	github.com/vmware/vic/lib/dns/_test/_obj_test/cache.go:128 +0x7a
github.com/vmware/vic/lib/dns.TestConcurrency.func2()
	/drone/src/github.com/vmware/vic/lib/dns/cache_test.go:190 +0xbe
created by github.com/vmware/vic/lib/dns.TestConcurrency
	/drone/src/github.com/vmware/vic/lib/dns/cache_test.go:210 +0x3e5

goroutine 2215823 [semacquire]:
sync.runtime_Semacquire(0xc82156623c)
	/usr/local/go/src/runtime/sema.go:47 +0x26
sync.(*Mutex).Lock(0xc821566238)
	/usr/local/go/src/sync/mutex.go:83 +0x1c4
sync.(*Pool).Put(0x1df2910, 0x12ad080, 0xc8add19450)
	/usr/local/go/src/sync/pool.go:81 +0xa5
fmt.(*pp).free(0xc8add19450)
	/usr/local/go/src/fmt/print.go:149 +0x79
fmt.Sprintf(0x12c8ba8, 0x5, 0xc8ba9d0ea0, 0x2, 0x2, 0x0, 0x0)
	/usr/local/go/src/fmt/print.go:205 +0xc8
github.com/vmware/vic/lib/dns.generateKey(0xc8201d3100, 0x1e, 0x10001, 0x0, 0x0)
	github.com/vmware/vic/lib/dns/_test/_obj_test/cache.go:79 +0x1aa
github.com/vmware/vic/lib/dns.(*Cache).Get(0xc8201afa40, 0xc8201d1290, 0xc8201d45e0)
	github.com/vmware/vic/lib/dns/_test/_obj_test/cache.go:128 +0x7a
github.com/vmware/vic/lib/dns.TestConcurrency.func2()
	/drone/src/github.com/vmware/vic/lib/dns/cache_test.go:190 +0xbe
created by github.com/vmware/vic/lib/dns.TestConcurrency
	/drone/src/github.com/vmware/vic/lib/dns/cache_test.go:210 +0x3e5

goroutine 16703 [semacquire]:
sync.runtime_Semacquire(0xc8215667bc)
	/usr/local/go/src/runtime/sema.go:47 +0x26
sync.(*Mutex).Lock(0xc8215667b8)
	/usr/local/go/src/sync/mutex.go:83 +0x1c4
sync.(*Pool).Get(0x1df2910, 0x0, 0x0)
	/usr/local/go/src/sync/pool.go:108 +0xa1
fmt.newPrinter(0x10)
	/usr/local/go/src/fmt/print.go:133 +0x27
fmt.Sprintf(0x12c8ba8, 0x5, 0xc83506dea0, 0x2, 0x2, 0x0, 0x0)
	/usr/local/go/src/fmt/print.go:202 +0x2b
github.com/vmware/vic/lib/dns.generateKey(0xc8201d3240, 0x1e, 0x1001c, 0x0, 0x0)
	github.com/vmware/vic/lib/dns/_test/_obj_test/cache.go:79 +0x1aa
github.com/vmware/vic/lib/dns.(*Cache).Get(0xc8201afa40, 0xc8201d1440, 0xc92fcdea20)
	github.com/vmware/vic/lib/dns/_test/_obj_test/cache.go:128 +0x7a
github.com/vmware/vic/lib/dns.TestConcurrency.func2()
	/drone/src/github.com/vmware/vic/lib/dns/cache_test.go:190 +0xbe
created by github.com/vmware/vic/lib/dns.TestConcurrency
	/drone/src/github.com/vmware/vic/lib/dns/cache_test.go:210 +0x3e5

goroutine 3973094 [GC assist wait]:
github.com/vmware/vic/lib/dns.generateKey(0xc8201d3100, 0x1e, 0x10001, 0x0, 0x0)
	github.com/vmware/vic/lib/dns/_test/_obj_test/cache.go:79 +0xf6
github.com/vmware/vic/lib/dns.(*Cache).Get(0xc8201afa40, 0xc8201d1290, 0xc8201d45e0)
	github.com/vmware/vic/lib/dns/_test/_obj_test/cache.go:128 +0x7a
github.com/vmware/vic/lib/dns.TestConcurrency.func2()
	/drone/src/github.com/vmware/vic/lib/dns/cache_test.go:190 +0xbe
created by github.com/vmware/vic/lib/dns.TestConcurrency
	/drone/src/github.com/vmware/vic/lib/dns/cache_test.go:210 +0x3e5

goroutine 110797 [semacquire]:
sync.runtime_Semacquire(0xc8215667bc)
	/usr/local/go/src/runtime/sema.go:47 +0x26
sync.(*Mutex).Lock(0xc8215667b8)
	/usr/local/go/src/sync/mutex.go:83 +0x1c4
sync.(*Pool).Put(0x1df2910, 0x12ad080, 0xc9094a7520)
	/usr/local/go/src/sync/pool.go:81 +0xa5
fmt.(*pp).free(0xc9094a7520)
	/usr/local/go/src/fmt/print.go:149 +0x79
fmt.Sprintf(0x12c8ba8, 0x5, 0xc8a922dea0, 0x2, 0x2, 0x0, 0x0)
	/usr/local/go/src/fmt/print.go:205 +0xc8
github.com/vmware/vic/lib/dns.generateKey(0xc8201d3160, 0x1e, 0x1001c, 0x0, 0x0)
	github.com/vmware/vic/lib/dns/_test/_obj_test/cache.go:79 +0x1aa
github.com/vmware/vic/lib/dns.(*Cache).Get(0xc8201afa40, 0xc8201d1320, 0xc83c206870)
	github.com/vmware/vic/lib/dns/_test/_obj_test/cache.go:128 +0x7a
github.com/vmware/vic/lib/dns.TestConcurrency.func2()
	/drone/src/github.com/vmware/vic/lib/dns/cache_test.go:190 +0xbe
created by github.com/vmware/vic/lib/dns.TestConcurrency
	/drone/src/github.com/vmware/vic/lib/dns/cache_test.go:210 +0x3e5
```

### PR DESCRIPTION
**Pull request by caglar10ur**
Hopefully fixes #1755

### CODE REVIEW

File: infra/scripts/coverage.sh [hunk 1]
```diff
@@ -44,6 +44,7 @@ generate_pkg_cover_data() {
 
     for pkg in "$@"; do
         f="$workdir/$(echo $pkg | tr / -).cover"
+        go test -v -i -covermode="$mode" -coverprofile="$f" "$pkg"
```
Comment (fdawg4l) [inline infra/scripts/coverage.sh]: 2 go test invocations for the same pkg.  Does that mean the tests will run twice now?

**Comment by fdawg4l** [comment]
LGTM.  `-i` doesn't run the test.  RTFM @fdawg4l.  :-)

**Comment by dougm** [comment]
lgtm

**Comment by caglar10ur** [comment]
♻️

**Pull request merged**